### PR TITLE
[PVR] Krypton: Fix PVR Manager crash on startup

### PR DESCRIPTION
**Pull request by ksooo**
Backport of #11283 

@Jalle19 for review?

### CODE REVIEW

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/pvr/PVRManager.h [hunk 1]
```diff
@@ -706,6 +706,8 @@ namespace PVR
     ManagerState                    m_managerState;
     std::unique_ptr<CStopWatch>     m_parentalTimer;
 
+    CCriticalSection                m_initMutex; // mutex for protecting pvr manager's start/restart sequence */
```
Comment (FrankHeimes) [inline xbmc/pvr/PVRManager.h]: `m_startStopMutex` would be a more appropriate name as it serializes only these two function.

**Review by FrankHeimes — CHANGES_REQUESTED** [review]
Acquiring two locks in a function is error-prone. It can work if the order of aquisition is always the same.
Can you guarantee that neither `Start()` nor `Stop()` is ever called while `m_critSection` is held?

**Comment by ksooo** [comment]
> Can you guarantee that neither Start() nor Stop() is ever called while m_critSection is held?

 As long as I'm maintaining this code and reviewing the patches, yes, because I'm very well aware of this problem.

 But, you are right, this code is not in good shape. The whole pvr manager start/stop logic is subject to complete rework, but not for Krypton.

File: xbmc/pvr/PVRManager.h (hunk 1 above)
Comment (ksooo) [inline xbmc/pvr/PVRManager.h]: changed. thx

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
I think this one is good to go. Well tested etc pp blah blah... (would have merged the respective master PR first, if jenkins would not act up on me currently)

**Pull request merged**